db/osqlcomm.c

```diff
@@ -6736,13 +6736,13 @@ int osql_process_packet(struct ireq *iq, unsigned long long rqid, uuid_t uuid,
         if (rc != 0) {
             if (err->errcode == OP_FAILED_UNIQ) {
                 if (iq->vfy_idx_track == 1 && iq->dup_key_insert == 1) {
-                    rc = ERR_UNCOMMITABLE_TXN;
-                    reqerrstr(iq, COMDB2_CSTRT_RC_DUP, "add key constraint "
-                                                   "duplicate key '%s' on "
-                                                   "table '%s' index %d",
+                    rc = ERR_UNCOMMITTABLE_TXN;
+                    reqerrstr(iq, COMDB2_CSTRT_RC_DUP, "Transaction is uncommittable: "
```

Comment:
The error code set in `reqerrstr` is normally sent to the replicant. But if it is set during a retry (as it is in uncommittable transaction cases), then this code gets overwritten with the rc [here](https://github.com/bloomberg/comdb2/blob/a39df61051770acc5167d2feefd4cbcbfe1be034/db/sltdbt.c#L444).

This is why the client ends up with `CDB2_VERIFY_ERROR` when they attempt an uncommittable transaction.